Plugin 1.2.4 | Script 1.3.0

Itokoyamato opened the pull request:
Pull requests:
-
PR #8 by mkafrin
- Optimize distance calculation
- Add config option `headingType` to use camera heading rather than ped heading for stereo positional audio
- Improve positional audio by using the head position, rather than ped position
---
PR #14  by Frazzle
- Optimize some loops
- Switch in-game data sharing from using in-game name to serverId
- Add TS name prefix config option `localNamePrefix`
---
Additional changes:
-
- Switch TS client matching from in-game name to TS UUID
- Improve positional stereo audio slightly
- Add config option to toggle stereo audio `enableStereoAudio`
- Add whitelisted TS channels config option `TSChannelWhitelist`
Screen will not be blocked in those specified channels
- Add TS name config option `localName` (Can be anything since we're using UUID now)
- Refactor TS plugin update checker with support for more customized informations
- Refactor JS script slightly
- Reworked script's radio system: simpler code, less network usage, less data on client & fix perf/leak
- Fixed some crashes and option issues on the plugin

A lot of code has been affected, some proper testing will be required.
I don't think I will be adding too much more to this PR. Better to validate those changes first and then move on new changes.

mkafrin commented:
Just for the sake of clarity, 

> Add config option headingType to use camera position rather than ped position for stereo positional audio

should probably be 

> Add config option headingType to use camera heading rather than ped heading for stereo positional audio

Itokoyamato commented:
@mkafrin Agreed, edited

Itokoyamato commented:
Fixed some crashes related to the changes and reworked the script's radio system: simpler code, less network usage, less data on client & fix some perf/leaks
I've done some testing with just one another player which went alright, will try to get some testing with more players tomorrow.

PichotM commented:
Something that I planned to do was to get rid of unnecessary calls of ts3Functions.getCurrentServerConnectionHandlerID in ts3_plugin
I believe the server connection handler id does not change so it could help optimization

Itokoyamato commented:
Agreed, it should be declared once at the beginning of functions using it

RussianRonin commented on fivem_script/tokovoip_script/src/c_utils.lua:
```diff
@@ -182,9 +182,9 @@ end
+	for i = 0, 256 do
 		if (NetworkIsPlayerActive(i)) then
-			players[#players + 1] = i;
+			table.insert(players, i);
```
```suggestion
			players[#players + 1] = i;
```

This should be swapped back to this because it is more performant. You can see the performance differences here: https://springrts.com/wiki/Lua_Performance#TEST_12:_Adding_Table_Items_.28table.insert_vs._.5B_.5D.29

Itokoyamato replied:
I thought so yeah, not sure why it was changed in the PR. Thanks

PichotM replied:
if you are worried about tiny optimization
you should also put almost everything as local
function & natives

wubzz commented on fivem_script/tokovoip_script/src/s_utils.lua:
```diff
@@ -16,20 +16,15 @@
-				break;
-			end
-		end
+		TriggerClientEvent("Tokovoip:setPlayerData", playerServerId, playerName, key, data);
```
Shouldn't `playerName` be `playerServerId` here given the listener in `c_utils.lua`?

`setPlayerData(playerServerId, key, data, shared)`
